File: software/examples/delphi/ExampleFindBorders.pas
program ExampleFindCorners;

{$ifdef MSWINDOWS}{$apptype CONSOLE}{$endif}
{$ifdef FPC}{$mode OBJFPC}{$H+}{$endif}

uses
  SysUtils, IPConnection, BrickletJoystick;

type
  TExample = class
  private
    ipcon: TIPConnection;
    js: TBrickletJoystick;
  public
    procedure ReachedCB(sender: TBrickletJoystick; const x: smallint; const y: smallint);
    procedure Execute;
  end;

const
  HOST = 'localhost';
  PORT = 4223;
  UID = '9Bw'; { Change to your UID }

var
  e: TExample;

{ Callback for x and y position outside of [-99..99] }
procedure TExample.ReachedCB(sender: TBrickletJoystick; const x: smallint; const y: smallint);
begin
  if (y = 100) then begin
    WriteLn('Top');
  end
  else if (y = -100) then begin
    WriteLn('Bottom');
  end;
  if (x = 100) then begin
    WriteLn('Right');
  end
  else if (x = -100) then begin
    WriteLn('Left');
  end;
  WriteLn('');
end;

procedure TExample.Execute;
begin
  { Create IP connection }
  ipcon := TIPConnection.Create;

  { Create device object }
  js := TBrickletJoystick.Create(UID, ipcon);

  { Connect to brickd }
  ipcon.Connect(HOST, PORT);
  { Don't use device before ipcon is connected }

  { Get threshold callbacks with a debounce time of 0.2 seconds (200ms) }
  js.SetDebouncePeriod(200);

  { Register threshold reached callback to procedure ReachedCB }
  js.OnPositionReached := {$ifdef FPC}@{$endif}ReachedCB;

  { Configure threshold for "x or y value outside of [-99..99]" }
  js.SetPositionCallbackThreshold('o', -99, 99, -99, 99);

  WriteLn('Press key to exit');
  ReadLn;
  ipcon.Destroy;
end;

begin
  e := TExample.Create;
  e.Execute;
  e.Destroy;
end.

File: software/examples/vbnet/ExampleFindBorders.vb
Imports System
Imports Tinkerforge

Module ExampleFindBorders
    Const HOST As String = "localhost"
    Const PORT As Integer = 4223
    Const UID As String = "XYZ" ' Change XYZ to the UID of your Joystick Bricklet

    ' Callback subroutine for position reached callback
    Sub PositionReachedCB(ByVal sender As BrickletJoystick, ByVal x As Short, _
                          ByVal y As Short)
        If y = 100 Then
            Console.WriteLine("Top")
        Else If y = -100 Then
            Console.WriteLine("Bottom")
        End If

        If x = 100 Then
            Console.WriteLine("Right")
        Else If x = -100 Then
            Console.WriteLine("Left")
        End If

        Console.WriteLine("")
    End Sub

    Sub Main()
        Dim ipcon As New IPConnection() ' Create IP connection
        Dim j As New BrickletJoystick(UID, ipcon) ' Create device object

        ipcon.Connect(HOST, PORT) ' Connect to brickd
        ' Don't use device before ipcon is connected

        ' Get threshold callbacks with a debounce time of 0.2 seconds (200ms)
        j.SetDebouncePeriod(200)

        ' Register position reached callback to subroutine PositionReachedCB
        AddHandler j.PositionReached, AddressOf PositionReachedCB

        ' Configure threshold for position "outside of -99, -99 to 99, 99"
        j.SetPositionCallbackThreshold("o"C, -99, 99, -99, 99)

        Console.WriteLine("Press key to exit")
        Console.ReadLine()
        ipcon.Disconnect()
    End Sub
End Module
